lib/client.js

```diff
@@ -75,8 +82,22 @@ function messageHandler(client, event) {
       pendingPromise.resolve(data.result);
     }
   } else if (ZAF_EVENT.test(data.key)) {
-    var key = data.key.replace(ZAF_EVENT, '');
-    if (client) {
+    var key = data.key.replace(ZAF_EVENT, ''),
+        msg = { appGuid: client._appGuid };
+    if (data.reply) {
```

Comment:
nitpick: `needsReply` would make more sense to me when reading, (so as not to get confused between the request and the reply itself), but no need to change since that would require a ZAF change as well.